cloudbuild.yaml

```diff
@@ -291,27 +291,32 @@ steps:
       set -o pipefail
       pids=()
       cloudProducts=("generic" "gke-autopilot")
+      declare -A versionsAndRegions=( [1.23]=us-east1 [1.24]=us-west1 [1.25]=us-central1 )
+
       for cloudProduct in ${cloudProducts[@]}
       do
-        if [ $cloudProduct = generic ]
-        then 
-          featureWithGate="CustomFasSyncInterval=false&SafeToEvict=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&PodHostname=true&SplitControllerAndExtensions=true&FleetAllocationOverflow=true&Example=true"
-          featureWithoutGate=""
-          testClusterLocation="us-west1-c"
-          testCluster="e2e-test-cluster"
-        else
-          featureWithGate="CustomFasSyncInterval=false&SafeToEvict=true&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&PodHostname=true&SplitControllerAndExtensions=true&FleetAllocationOverflow=true&Example=true"
-          featureWithoutGate="SplitControllerAndExtensions=true"
-          testClusterLocation="us-west1"
-          testCluster="gke-autopilot-e2e-test-cluster-1-24"
-        fi
-        # TODO: Add more versions in the list, and generate the test cluster name in the loop rather than hardcode once the clusters are created
-        k8sVersions=("1.24")
-        for k8sVersion in ${k8sVersions[@]}
+        for version in "${!versionsAndRegions[@]}"
         do
+          region=${versionsAndRegions[$version]}
+          if [ $cloudProduct = gke-autopilot ] && [ $version = 1.23 ]
+          then
+            continue
+          fi
+          if [ $cloudProduct = generic ]
+          then
+            featureWithGate="CustomFasSyncInterval=false&SafeToEvict=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&PodHostname=true&SplitControllerAndExtensions=true&FleetAllocationOverflow=true&Example=true"
+            featureWithoutGate=""
+            testClusterLocation="${region}-c"
+            testCluster="gke-standard-e2e-test-cluster-${version//./-}"
```

Comment:
nit: Maybe something to consider in the future would be to name the clusters as `"${cloudProduct}-e2e-test-cluster-${version//./-}"`. So we'd have names like `generic-e2e-test-cluster-1-24` and `gke-autopilot-e2e-test-cluster-1-24`. 

I'm thinking that once a couple of feature gates get promoted to beta, we won't need the if/else here for feature gates based on the provider, and if we move to regional standard clusters then the only difference remaining is how we construct the test cluster names.